setup.py

```diff
@@ -726,37 +715,11 @@ def _populate_hdf5_info(dirstosearch, inc_dirs, libs, lib_dirs):
     sys.stdout.write('NETCDF_PLUGIN_DIR not set, no netcdf compression plugins installed\n')
     data_files = []
 
-setup(name="netCDF4",
-      cmdclass=cmdclass,
-      version=extract_version(netcdf4_src_pyx),
-      long_description="netCDF version 4 has many features not found in earlier versions of the library, such as hierarchical groups, zlib compression, multiple unlimited dimensions, and new data types.  It is implemented on top of HDF5.  This module implements most of the new features, and can read and write netCDF files compatible with older versions of the library.  The API is modelled after Scientific.IO.NetCDF, and should be familiar to users of that module.\n\nThis project is hosted on a `GitHub repository <https://github.com/Unidata/netcdf4-python>`_ where you may access the most up-to-date source.",
-      author="Jeff Whitaker",
-      author_email="jeffrey.s.whitaker@noaa.gov",
-      url="http://github.com/Unidata/netcdf4-python",
-      download_url="http://python.org/pypi/netCDF4",
-      platforms=["any"],
-      license='License :: OSI Approved :: MIT License',
-      description="Provides an object-oriented python interface to the netCDF version 4 library.",
-      keywords=['numpy', 'netcdf', 'data', 'science', 'network', 'oceanography',
-                'meteorology', 'climate'],
-      classifiers=["Development Status :: 3 - Alpha",
-                   "Programming Language :: Python :: 3",
-                   "Programming Language :: Python :: 3.7",
-                   "Programming Language :: Python :: 3.8",
-                   "Programming Language :: Python :: 3.9",
-                   "Programming Language :: Python :: 3.10",
-                   "Programming Language :: Python :: 3.11",
-                   "Intended Audience :: Science/Research",
-                   "License :: OSI Approved :: MIT License",
-                   "Topic :: Software Development :: Libraries :: Python Modules",
-                   "Topic :: System :: Archiving :: Compression",
-                   "Operating System :: OS Independent"],
-      packages=find_namespace_packages(where="src"),
-      package_dir={'':'src'},
-      package_data={"netCDF4.plugins": ["lib__nc*"]},
-      ext_modules=ext_modules,
-      python_requires=">=3.7",
-      **setuptools_extra_kwargs)
+# See pyproject.toml for project metadata
+setup(
+    version=extract_version(netcdf4_src_pyx),
```

Comment:
I'm sorry I'm late here but I'd like to keep at least the name metadata in the `setup.py` for a while. The reason is b/c GitHub doesn't parse pyproject.toml yet and we lose the dependency graph info in https://github.com/Unidata/netcdf4-python/network/dependents without it.

https://github.com/community/community/discussions/6456

Reply:
that seems like a GH bug. Besides discovering other connected projects, what is a dependency graph used for?

Reply:
reading comments from the past two weeks, it appears that the Dependency Graph team are investigating a fix. I'd expect to see this resolved soon, particularly since the PEP is now supported by setuptools.

Reply:
It has been years since that was originally reported. I would not hold my breath. Also, re-adding just the name metadata in the setup.py is not that hard.

Reply:
we could simply plop the relevant metadata to setup.py until it's resolved; see e.g. [flask](https://github.com/pallets/flask/blob/cc66213e579d6b35d9951c21b685d0078f373c44/setup.py#L3-L17). I'm in the middle of refactoring setup.py, so I can include that when ready.

Reply:
That is what I asked in my first message. Sorry if that was not clear.

Reply:
yup, no probs I'll do that in that forthcoming PR, it just might be a few more days without a dependency graph.